infra/azure/terraform/capz/role-assignments/main.tf

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+variable "resource_group_name" {
+  type = string
+}
+
+variable "container_registry_scope" {
+  type = string
+}
+
+variable "storage_account_scope" {
+  type = string
+}
+
+variable "subscription_id" {
+  type = string
+}
+
+data "azuread_service_principal" "az_service_principal" {
+  object_id = "8eb97873-17bb-46a5-aed9-1967d83ca7ca"
+}

```

Comment:
Maybe this should be by name instead of object id?  That way it can be re-used with out knowing the object id